apps/hash-frontend/src/lib/user-and-org.ts

```diff
@@ -248,10 +255,30 @@ export const constructOrg = (params: {
       types.linkEntityType.orgMembership.linkEntityTypeId,
   ) as Entity<OrgMembershipProperties>[];
 
+  const avatarLinkAndEntities = getOutgoingLinkAndTargetEntities(
+    subgraph,
+    orgEntity.metadata.recordId.entityId,
+    intervalForTimestamp(new Date().toISOString() as Timestamp),
+  ).filter(
+    ({ linkEntity }) =>
+      linkEntity[0]?.metadata.entityTypeId ===
+      types.linkEntityType.hasAvatar.linkEntityTypeId,
+  );
+
+  const hasAvatar = avatarLinkAndEntities[0];
+
   const org = constructMinimalOrg({
     orgEntity,
   }) as Org;
 
+  org.hasAvatar = hasAvatar
+    ? {
+        // these are each arrays because each entity can have multiple revisions
+        linkEntity: hasAvatar.linkEntity[0] as LinkEntity,
+        rightEntity: hasAvatar.rightEntity[0] as unknown as RemoteImageFile,
```

Comment:
Just to improve my TS skills: whe `unknown` is needed here? Shouldn't the `Entity` type and `RemoteImageFile` type overlap?

Reply:
They do overlap, but they don't "sufficiently" overlap to satisfy the compiler that we should be doing this. The problem here is that `RemoteImageFile` has specific properties which a generic `Entity` doesn't, and therefore it is unsafe to do this cast, since there's no indication that `rightEntity` has any of the specific properties required